pkg/appgw/internaltypes.go

```diff
@@ -63,8 +65,8 @@ func getResourceKey(namespace, name string) string {
 	return fmt.Sprintf("%v/%v", namespace, name)
 }
 
-func generateHTTPSettingsName(serviceName string, servicePort string, backendPortNo int32) string {
-	return fmt.Sprintf("k8s-%v-%v-bp-%v", serviceName, servicePort, backendPortNo)
+func generateHTTPSettingsName(serviceName string, servicePort string, backendPortNo int32, ingress string) string {
+	return fmt.Sprintf("k8s-%v-%v-bp-%v-%s", serviceName, servicePort, backendPortNo, ingress)
```

Comment:
We were trying to find all appgw's configured as ingress using "contains "k8s-" query in config and it gave more matches since "k8s-" was also used by other users in their configs. Maybe having something unique "k8s-appgw-" or the likes will help query these auto-generated configs when ingress controller is used.